docs/src/prototypes/dropdowns/inputWithDropdown.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react'
+import * as _ from 'lodash'
+import keyboardKey from 'keyboard-key'
+import { Dropdown, DropdownProps } from '@stardust-ui/react'
+
+import { atMentionItems, AtMentionItem } from './dataMocks'
+import { insertTextAtCursorPosition } from './utils'
+import { CustomPortal } from './CustomPortal'
+
+interface InputWithDropdownState {
+  dropdownOpen?: boolean
+  searchQuery?: string
+}
+
+const editorStyle: React.CSSProperties = {
+  backgroundColor: 'lightgrey',
+  padding: '5px',
+  minHeight: '100px',
+  outline: 0,
+}
+
+class InputWithDropdown extends React.Component<{}, InputWithDropdownState> {
+  private readonly initialState: InputWithDropdownState = {
+    dropdownOpen: false,
+    searchQuery: '',
+  }
```

Comment:
NIT: `initialState` can be extracted to a variable as it doesn't have any coupling with the class

Reply:
agreed but it also doesn't make sense outside of this component... with that argument most of `readonly` members do not have any coupling usually with their class (are merely constants)